src/components/sidebar/ContextAwareSidebar.tsx

```diff
@@ -182,9 +182,15 @@ class ContextAwareSidebar
 
     persistence.fetchEdges(course.guid).then((edges) => {
 
-      const sources = edges.filter((edge) => {
-        return (this.stripId(edge.destinationId) === resource.id);
-      });
+      // returns a list of edges pointing to the current page, with no edges sharing the same source
+      const sources: Edge[] = edges.filter(edge => this.stripId(edge.destinationId) === resource.id)
+                                                       .reduce((prev: [any, Edge[]], edge) => {
```

Comment:
I know we banged this out quickly, but you might rename `prev` to "usedResources" or something like that to make it more obvious what's going on here. Another way you could approach it would be to use an object instead of a tuple for the accumulator. E.g. `{ usedResources: {}, edges: [] }` and then access the `.edges` property instead of `[1]` after reducing.

Reply:
made the change for accumulator to be an object, as suggested